### ISSUE TITLE
[API] Endpoint for adding/updating user info

### ISSUE THREAD

**Issue by Aksh-Bansal-dev** [BitByte-TPC/gymkhana#132]

**Comment by Samy-33** [comment]
@PranshuNayak, I am removing this assignment as there are few others assigned to you which you can work on and others can take these tasks up.

**Comment by arminpatel** [comment]
I'm working on this.

**Comment by arminpatel** [comment]
I'm a little confused which model is to be serialized for the endpoint. 
Should I create a serializer for only User model or for all of them (Student, Faculty and Staff)?
Or should I create serializer for Student, Faculty and Staff and add extra fields to them (like name, email etc)?
IMO second seems better

**Comment by Samy-33** [comment]
How about creating a serializer for User, and have three fields in it for student, faculty, staff.

This is nested serialization: https://www.django-rest-framework.org/api-guide/relations/#nested-relationships

**Comment by arminpatel** [comment]
> How about creating a serializer for User, and have three fields in it for student, faculty, staff.
> 
> This is nested serialization: https://www.django-rest-framework.org/api-guide/relations/#nested-relationships

Exactly what I wanted. Thanks.

**Comment by arminpatel** [comment]
Does this endpoint also creates users?

**Comment by Aksh-Bansal-dev** [comment]
Yeah, you can create 2 endpoints one for updating and one for creating.

**Comment by arminpatel** [comment]
> raise 2 separate CLs

what are CLs? 😅

**Comment by Samy-33** [comment]
> what are CLs? 😅

Sorry, got confused between internal terminology and external. It's called a changelist same as pull request.

So raise two PRs separately for update and create.

**Comment by arminpatel** [comment]
Who will have permission to update user?

**Comment by Aksh-Bansal-dev** [comment]
> Who will have permission to update user?

I think only the user should have permission to edit their profile details.

**Comment by Samy-33** [comment]
> I think only the user should have permission to edit their profile details.

Admin user should have permission to edit user information as well.
